modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItTxResourcesVacuumTest.java

```diff
@@ -0,0 +1,559 @@
+            .set("val", "" + (t.longValue("key") + 1));
+
+    private static final int REPLICAS = 2;
+
+    /** Nodes bootstrap configuration pattern. */
+    private static final String NODE_BOOTSTRAP_CFG_TEMPLATE = "{\n"
+            + "  network: {\n"
+            + "    port: {},\n"
+            + "    nodeFinder: {\n"
+            + "      netClusterNodes: [ {} ]\n"
+            + "    }\n"
+            + "  },\n"
+            + "  clientConnector: { port:{} },\n"
+            + "  rest.port: {},\n"
+            + "  raft: { responseTimeout: 30000 },"
+            + "  compute.threadPoolSize: 1\n"
+            + "}";
+
+    private ExecutorService txStateStorageExecutor = Executors.newSingleThreadExecutor();
+
+    @BeforeEach
+    @Override
+    public void setup(TestInfo testInfo) throws Exception {
+        super.setup(testInfo);
+
+        String zoneSql = "create zone test_zone with partitions=20, replicas=" + REPLICAS
+                + ", storage_profiles='" + DEFAULT_STORAGE_PROFILE + "'";
+        String sql = "create table " + TABLE_NAME + " (key bigint primary key, val varchar(20)) with primary_zone='TEST_ZONE'";
+
+        cluster.doInSession(0, session -> {
+            executeUpdate(zoneSql, session);
+            executeUpdate(sql, session);
+        });
+
+        txStateStorageExecutor = Executors.newSingleThreadExecutor(IgniteThreadFactory.create("test", "tx-state-storage-test-pool", log,
+                ThreadOperation.STORAGE_READ));
+    }
+
+    @Override
+    @AfterEach
+    public void tearDown() {
+        shutdownAndAwaitTermination(txStateStorageExecutor, 10, TimeUnit.SECONDS);
+
+        super.tearDown();
+    }
+
+    @Override
+    protected void customizeInitParameters(InitParametersBuilder builder) {
+        super.customizeInitParameters(builder);
+
+        builder.clusterConfiguration("{"
+                + "  transaction: {"
+                + "      implicitTransactionTimeout: 30000,"
+                + "      txnResourceTtl: 2"
+                + "  },"
+                + "  replication: {"
+                + "      rpcTimeout: 30000"
+                + "  },"
+                + "}");
+    }
+
+    @Override
+    protected int initialNodes() {
+        return 3;
+    }
+
+    /**
+     * Returns node bootstrap config template.
+     *
+     * @return Node bootstrap config template.
+     */
+    @Override
+    protected String getNodeBootstrapConfigTemplate() {
+        return NODE_BOOTSTRAP_CFG_TEMPLATE;
+    }
+
+    @Test
+    public void testVacuum() throws InterruptedException {
+        IgniteImpl node = anyNode();
+
+        RecordView<Tuple> view = node.tables().table(TABLE_NAME).recordView();
+
+        // Put some value into the table.
+        Transaction tx = node.transactions().begin();
+        UUID txId = txId(tx);
+
+        log.info("Test: Loading the data [tx={}].", txId);
+
+        Tuple tuple = findTupleToBeHostedOnNode(node, TABLE_NAME, tx, INITIAL_TUPLE, NEXT_TUPLE, true);
```

Comment:
Why so complicated? Do we really need to find specific node within given scenario?

Reply:
I believe so, for example, there are the scenarios when we have one value that is going to be hosted on 2 replicas of 2 nodes of 3, and w should generate some value that would be hosted on 3rd node. At least, having the utility method that is able to generate such value is the easiest way to do so, isn't it?

Reply:
Argh, pfff, mmm